File: src/main/java/info/skyblond/archivedag/config/JwtConfig.java
package info.skyblond.archivedag.config;

import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.authentication.AccountStatusUserDetailsChecker;

@Configuration
@EnableConfigurationProperties(JwtProperties.class)
public class JwtConfig {
    @Bean
    public AccountStatusUserDetailsChecker accountStatusUserDetailsChecker() {
        return new AccountStatusUserDetailsChecker();
    }
}

File: backend/src/main/kotlin/info/skyblond/archivedag/arudaz/config/JwtConfig.kt
package info.skyblond.archivedag.arudaz.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.authentication.AccountStatusUserDetailsChecker

@Configuration
class JwtConfig {
    /**
     * For JWT token resolve. Check user status.
     * This cannot be merged into SpringSecurityConfig, it will be a loop.
     * */
    @Bean
    fun accountStatusUserDetailsChecker(): AccountStatusUserDetailsChecker {
        return AccountStatusUserDetailsChecker()
    }
}
